napari/animation/animate.py

```diff
@@ -0,0 +1,344 @@
+                frame
+            ][j]
+
+        # adjust slider positions
+        for i in range(self.interpolated_states["sliders"].shape[1]):
+            self.viewer.dims.set_point(
+                i, self.interpolated_states["sliders"][frame][i]
+            )
+
+        # update view
+        self.viewer.window.qt_viewer.view.camera.view_changed()
+
+    def frame_generator(self, frame=None, with_viewer=False, max_shape=None):
+        """Generator of frames of the animation
+
+        Parameters
+        -------
+        frame : int
+            Specific frame to return.
+        with_viewer : bool
+            If True includes the napari viewer, otherwise just includes the
+            canvas.
+        max_shape : int
+            Size to which to rescale the largest axis. Only used without
+            viewer (with_viewer = False).
+
+        """
+
+        # create states
+        self.create_steps()
+
+        # capture SceneCanvas size of frist frame to set size of next ones
+        self.update_viewer_from_state(0)
+        frame_size = self.viewer.window.qt_viewer.canvas.size
+        # if no max size provided, take current size as default
+        if max_shape is None:
+            max_shape = max(frame_size)
+
+        # return specific frame
+        if frame is not None:
+            self.update_viewer_from_state(frame)
+            image = self.viewer.screenshot(
+                max_shape=max_shape, with_viewer=with_viewer
+            )
+            while True:
+                yield image
+        # return all frames as a generator
+        else:
+            for i in range(len(self.states_dict)):
+                self.update_viewer_from_state(i)
+                if not with_viewer:
+                    self.viewer.window.qt_viewer.canvas.size = frame_size
+                yield self.viewer.screenshot(
+                    max_shape=max_shape, with_viewer=with_viewer
+                )
+
+    def animate(
+        self,
+        name="movie.mp4",
+        fps=20,
+        quality=5,
+        format=None,
+        with_viewer=False,
+        max_shape=None,
+    ):
+        """Create a movie based on key-frames
+
+        Parameters
+        -------
+        name : str
+            name to use for saving the movie (can also be a path)
+            should be either .mp4 or .gif
+        fps : int
+            frames per second
+        quality: float
+            number from 1 (lowest quality) to 9
+            only applies to mp4
+        format: str
+            The format to use to write the file. By default imageio selects the appropriate for you based on the filename.
+        with_viewer : bool
+            If True includes the napari viewer, otherwise just includes the
+            canvas.
+        max_shape : int
+            Size to which to rescale the largest axis. Only used without
+            viewer (with_viewer = False).
+        """
+
```

Comment:
I think here we might need to add the lines
```python
if shape is None:
    shape = self.viewer.screenshot(with_viewer=with_viewer)[:2]
```
and then use `shape` inside the `frame_generator`.
Where not since my comment about using [skimage resize](https://scikit-image.org/docs/stable/api/skimage.transform.html#skimage.transform.resize) in the screenshot method I'm now talking `shape` instead of just `max_shape`